File: library/src/main/java/me/proxer/library/api/notifications/InternalApi.java
package me.proxer.library.api.notifications;

import me.proxer.library.api.ProxerCall;
import me.proxer.library.entity.notifications.NewsArticle;
import me.proxer.library.entity.notifications.Notification;
import me.proxer.library.entity.notifications.NotificationInfo;
import me.proxer.library.enums.NotificationFilter;
import retrofit2.http.*;

import javax.annotation.ParametersAreNullableByDefault;
import java.util.List;

/**
 * @author Ruben Gees
 */
@ParametersAreNullableByDefault
interface InternalApi {

    @GET("notifications/news")
    ProxerCall<List<NewsArticle>> news(@Query("p") Integer page,
                                       @Query("limit") Integer limit,
                                       @Query("set_read") Boolean markAsRead);

    @GET("notifications/notifications")
    ProxerCall<List<Notification>> notifications(@Query("p") Integer page,
                                                 @Query("limit") Integer limit,
                                                 @Query("set_read") Boolean markAsRead,
                                                 @Query("filter") NotificationFilter filter);

    @GET("notifications/count")
    ProxerCall<NotificationInfo> notificationInfo();

    @FormUrlEncoded
    @POST("notifications/delete")
    ProxerCall<Void> deleteNotification(@Field("nid") String id);
}

File: library/src/main/kotlin/me/proxer/library/api/notifications/InternalApi.kt
package me.proxer.library.api.notifications

import me.proxer.library.ProxerCall
import me.proxer.library.entity.notifications.NewsArticle
import me.proxer.library.entity.notifications.Notification
import me.proxer.library.entity.notifications.NotificationInfo
import me.proxer.library.enums.NotificationFilter
import retrofit2.http.Field
import retrofit2.http.FormUrlEncoded
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Query

/**
 * @author Ruben Gees
 */
@Suppress("UndocumentedPublicFunction")
internal interface InternalApi {

    @GET("notifications/news")
    fun news(
        @Query("p") page: Int?,
        @Query("limit") limit: Int?,
        @Query("set_read") markAsRead: Boolean?
    ): ProxerCall<List<NewsArticle>>

    @GET("notifications/notifications")
    fun notifications(
        @Query("p") page: Int?,
        @Query("limit") limit: Int?,
        @Query("set_read") markAsRead: Boolean?,
        @Query("filter") filter: NotificationFilter?
    ): ProxerCall<List<Notification>>

    @GET("notifications/count")
    fun notificationInfo(): ProxerCall<NotificationInfo>

    @FormUrlEncoded
    @POST("notifications/delete")
    fun deleteNotification(@Field("nid") id: String?): ProxerCall<Unit>
}
